exchanges/request/limit.go

```diff
@@ -26,23 +30,43 @@ const (
 // BasicLimit denotes basic rate limit that implements the Limiter interface
 // does not need to set endpoint functionality.
 type BasicLimit struct {
-	r *rate.Limiter
+	r *RateLimiterWithToken
 }
 
 // Limit executes a single rate limit set by NewRateLimit
-func (b *BasicLimit) Limit(ctx context.Context, _ EndpointLimit) error {
-	return b.r.Wait(ctx)
+func (b *BasicLimit) Limit(context.Context, EndpointLimit) (*RateLimiterWithToken, error) {
```

Comment:
Is there any need for these params?

Reply:
There is no need for even the struct. 👻 I shall burn it with fire. 🔥